include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,145 @@
+ * \implements seqan3::TriviallyCopyable
+ * \implements seqan3::StandardLayout
+ * \implements std::Regular
+ *
+ * \details
+ *
+ * This alphabet represents a unit in a CIGAR string, typically found in the
+ * SAM and BAM formats. It consists of a number and a seqan3::cigar_op symbol.
+ *
+ * The "char representation" of this alphabet is a seqan3::small_string. This
+ * has some implications when using it in code that doesn't expect a range
+ * of symbols.
+ *
+ * To avoid confusion between string and char literal, this alphabet has
+ * no user defined literal operators. Always assign from a pair of
+ * uint32_t and seqan3::cigar_op.
+ *
+ * ### Example
+ *
+ * \include test/snippet/alphabet/cigar/cigar.cpp
+ */
+class cigar : public alphabet_tuple_base<cigar, uint32_t, cigar_op>
+{
+private:
+    //!\brief The base class.
+    using base_t = alphabet_tuple_base<cigar, uint32_t, cigar_op>;
+
+    //!\cond \brief Befriend seqan3::alphabet_base.
+    friend base_t;
+    //!\endcond
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr cigar()                          noexcept = default; //!< Defaulted.
+    constexpr cigar(cigar const &)             noexcept = default; //!< Defaulted.
+    constexpr cigar(cigar &&)                  noexcept = default; //!< Defaulted.
+    constexpr cigar & operator=(cigar const &) noexcept = default; //!< Defaulted.
+    constexpr cigar & operator=(cigar &&)      noexcept = default; //!< Defaulted.
+    ~cigar()                                   noexcept = default; //!< Defaulted.
+
+    // Inherit constructors from base
+    using base_t::base_t;
+
+    //!\copydoc alphabet_tuple_base::alphabet_tuple_base(component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar(component_type const alph) noexcept {} ))
+    //!\copydoc alphabet_tuple_base::alphabet_tuple_base(indirect_component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar(indirect_component_type const alph) noexcept {} ))
+    //!\copydoc alphabet_tuple_base::operator=(component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar & operator=(component_type const alph) noexcept {} ))
+    //!\copydoc alphabet_tuple_base::operator=(indirect_component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar & operator=(indirect_component_type const alph) noexcept {} ))
+    //!\}
+
+    // Inherit operators from base
+    using base_t::operator=;
+    using base_t::operator==;
+    using base_t::operator!=;
+    using base_t::operator>=;
+    using base_t::operator<=;
+    using base_t::operator<;
+    using base_t::operator>;
+
+    /*!\name Read functions
+     * \{
+     */
+    //!\brief Return the character representation.
+    small_string<11> to_char() const noexcept
+    {
+        small_string<11> ret{};
+        ret.resize(11);
+
+        auto [ ptr, errc ] = std::to_chars(ret.data(), ret.data() + 10, seqan3::to_rank(get<0>(*this)));
```

Comment:
why to_rank? if I have a tuple (5, 'M') I want it to print "5M", so I can directly take the count value without the rank?